mmcv/ops/nms.py

```diff
@@ -37,14 +50,26 @@ def symbolic(g, bboxes, scores, iou_threshold, offset):
             from torch.onnx.symbolic_opset9 import select, squeeze, unsqueeze
             boxes = unsqueeze(g, bboxes, 0)
             scores = unsqueeze(g, unsqueeze(g, scores, 0), 0)
-            max_output_per_class = g.op(
-                'Constant',
-                value_t=torch.tensor([sys.maxsize], dtype=torch.long))
+
+            from torch.onnx.symbolic_helper import _is_value, _size_helper
+            if not _is_value(max_num):
```

Comment:
Here `max_num` should be a constant int value. We could remove `if` block and only assert `max_num > 0` here.
```
assert max_num > 0, 'max_num should be greater than 0 while in exploring to ONNX.'
```

Reply:
It seems that after the changes in `tensorrt_utils.preprocess_onnx` before exporting to TensorRT, we can save the ability to define `max_num` not only as a constant int value.

Reply:
It's not necessary to set `max_num` to be dynamic for deployment. Memory allocation for tensorrt nms plugin highly depends on this parameter and it shouldn't be dynamic.

Reply:
Then what benefits it would bring if you keep `max_num` as dynamic for backend other than TensorRT?

Reply:
For example, this can be useful in the OpenVINO backend.
Support for the ability of `max_num` to be non-constant allows for model reshaping and improves performance.

Reply:
Then if you force to set `max_num`to a constant value and this value is much smaller than the `bbox.shape[0]`, won't it improve the performance much more? `max_num` is normally set to 100 in the config file.

Reply:
But by choosing too small a value for `max_num`, we can reduce the quality of the model.

Previously, calling `batched_nms` from RPNHead did not pass `max_num` and this caused some performance problems. To solve them, it was proposed to define `max_num` using `bbox.shape`.